Installation issue: UCX Compile Failed

There is a build error when I install ucx(almost all-versions)
And it's installed sucessfully with a earlier spack `commid-id`, such as:
```
commit e519bbadc3c55f6fafacadc4ae950fb02565fe27 (HEAD)
Author: Howard Pritchard <howardp@lanl.gov>
Date:   Fri Feb 28 10:22:37 2020 -0700

    UCX: add 1.7.0 (#15266)

    Signed-off-by: Howard Pritchard <howardp@lanl.gov>
```

spack `commid-id` now:
```
commit 0d96e8a27d13166049bb9280e2e270a568fe9759 (HEAD -> develop, origin/develop, origin/HEAD)
Author: tilne <55806862+tilne@users.noreply.github.com>
Date:   Thu Apr 9 20:00:03 2020 -0700

    update URL and sha256 for aws-parallelcluster 2.6.1 (#15980)

    Signed-off-by: Tim Lane <tilne@amazon.com>
```

Error log:
```
==> 22781: Installing ucx
==> Finding buildcaches in /home/spack_mirror/build_cache
==> Fetching file:///home/spack_mirror/build_cache/linux-centos8-aarch64-gcc-8.2.1-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs.spec.yaml
curl: (37) Couldn't open file /home/spack_mirror/build_cache/linux-centos8-aarch64-gcc-8.2.1-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs.spec.yaml
==> Failed to fetch file from URL: file:///home/spack_mirror/build_cache/linux-centos8-aarch64-gcc-8.2.1-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs.spec.yaml
    Curl failed with error 37
==> Fetching from file:///home/spack_mirror/build_cache/linux-centos8-aarch64-gcc-8.2.1-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs.spec.yaml failed.
==> Using cached archive: /home/xiaojun/spack/var/spack/cache/_source-cache/archive/6a/6ab81ee187bfd554fe7e549da93a11bfac420df87d99ee61ffab7bb19bdd3371.tar.gz
==> Staging archive: /tmp/root/spack-stage/spack-stage-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs/ucx-1.7.0.tar.gz
==> Created stage in /tmp/root/spack-stage/spack-stage-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs
==> No patches needed for ucx
==> 22781: ucx: Building ucx [AutotoolsPackage]
==> 22781: ucx: Executing phase: 'autoreconf'
==> 22781: ucx: Executing phase: 'configure'
==> 22781: ucx: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j4'

2 errors found in build log:
     587    /tmp/root/spack-stage/spack-stage-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs/spack-src/src/ucm/util/reloc.h:66:(.text+0x1e00): dangerous relocation: unsupported relocation
     588    /usr/local/bin/ld: malloc/.libs/libucm_la-malloc_hook.o: relocation R_AARCH64_ADR_PREL_PG_HI21 against symbol `ucm_global_opts' which may bind externally can not be used when making a shared
            object; recompile with -fPIC
     589    malloc/.libs/libucm_la-malloc_hook.o: in function `ucm_mmap_hook_mode':
     590    /tmp/root/spack-stage/spack-stage-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs/spack-src/src/ucm/mmap/mmap.h:45:(.text+0x1e7c): dangerous relocation: unsupported relocation
     591    /usr/local/bin/ld: malloc/.libs/libucm_la-malloc_hook.o: relocation R_AARCH64_ADR_PREL_PG_HI21 against symbol `__free_hook@@GLIBC_2.17' which may bind externally can not be used when making a
             shared object; recompile with -fPIC
     592    /usr/local/bin/ld: malloc/.libs/libucm_la-malloc_hook.o(.text+0x1ed8): unresolvable R_AARCH64_ADR_PREL_PG_HI21 relocation against symbol `__free_hook@@GLIBC_2.17'
  >> 593    /usr/local/bin/ld: final link failed: bad value
  >> 594    collect2: error: ld returned 1 exit status
     595    make[3]: *** [Makefile:696: libucm.la] Error 1
     596    make[3]: Leaving directory '/tmp/root/spack-stage/spack-stage-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs/spack-src/src/ucm'
     597    make[2]: *** [Makefile:886: all-recursive] Error 1
     598    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs/spack-src/src/ucm'
     599    make[1]: *** [Makefile:719: all-recursive] Error 1
     600    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs/spack-src'

See build log for details:
  /tmp/root/spack-stage/spack-stage-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs/spack-build-out.txt

==> Error: Failed to install ucx due to ChildError: ProcessError: Command exited with status 2:
    'make' '-j4'
2 errors found in build log:
     587    /tmp/root/spack-stage/spack-stage-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs/spack-src/src/ucm/util/reloc.h:66:(.text+0x1e00): dangerous relocation: unsupported relocation
     588    /usr/local/bin/ld: malloc/.libs/libucm_la-malloc_hook.o: relocation R_AARCH64_ADR_PREL_PG_HI21 against symbol `ucm_global_opts' which may bind externally can not be used when making a shared
            object; recompile with -fPIC
     589    malloc/.libs/libucm_la-malloc_hook.o: in function `ucm_mmap_hook_mode':
     590    /tmp/root/spack-stage/spack-stage-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs/spack-src/src/ucm/mmap/mmap.h:45:(.text+0x1e7c): dangerous relocation: unsupported relocation
     591    /usr/local/bin/ld: malloc/.libs/libucm_la-malloc_hook.o: relocation R_AARCH64_ADR_PREL_PG_HI21 against symbol `__free_hook@@GLIBC_2.17' which may bind externally can not be used when making a
             shared object; recompile with -fPIC
     592    /usr/local/bin/ld: malloc/.libs/libucm_la-malloc_hook.o(.text+0x1ed8): unresolvable R_AARCH64_ADR_PREL_PG_HI21 relocation against symbol `__free_hook@@GLIBC_2.17'
  >> 593    /usr/local/bin/ld: final link failed: bad value
  >> 594    collect2: error: ld returned 1 exit status
     595    make[3]: *** [Makefile:696: libucm.la] Error 1
     596    make[3]: Leaving directory '/tmp/root/spack-stage/spack-stage-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs/spack-src/src/ucm'
     597    make[2]: *** [Makefile:886: all-recursive] Error 1
     598    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs/spack-src/src/ucm'
     599    make[1]: *** [Makefile:719: all-recursive] Error 1
     600    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs/spack-src'

See build log for details:
  /tmp/root/spack-stage/spack-stage-ucx-1.7.0-fau66yhl4uxmq2575ltffgom7dqxd5gs/spack-build-out.txt
Traceback (most recent call last):
  File "/home/xiaojun/spack/lib/spack/spack/build_environment.py", line 801, in child_process
    return_value = function()
  File "/home/xiaojun/spack/lib/spack/spack/installer.py", line 1113, in build_process
    phase(pkg.spec, pkg.prefix)
  File "/home/xiaojun/spack/lib/spack/spack/package.py", line 108, in phase_wrapper
    phase(spec, prefix)
  File "/home/xiaojun/spack/lib/spack/spack/build_systems/autotools.py", line 290, in build
    inspect.getmodule(self).make(*self.build_targets)
  File "/home/xiaojun/spack/lib/spack/spack/build_environment.py", line 131, in __call__
    return super(MakeExecutable, self).__call__(*args, **kwargs)
  File "/home/xiaojun/spack/lib/spack/spack/util/executable.py", line 189, in __call__
    proc.returncode, long_msg)
spack.util.executable.ProcessError: Command exited with status 2:
    'make' '-j4'
```

--- a/var/spack/repos/builtin/packages/ucx/package.py
+++ b/var/spack/repos/builtin/packages/ucx/package.py
@@ -16,6 +16,7 @@ class Ucx(AutotoolsPackage):
     maintainers = ['hppritcha']
 
     # Current
+    version('1.8.0', sha256='e400f7aa5354971c8f5ac6b881dc2846143851df868088c37d432c076445628d')
     version('1.7.0', sha256='6ab81ee187bfd554fe7e549da93a11bfac420df87d99ee61ffab7bb19bdd3371')
     version('1.6.1', sha256='1425648aa03f5fa40e4bc5c4a5a83fe0292e2fe44f6054352fbebbf6d8f342a1')
     version('1.6.0', sha256='360e885dd7f706a19b673035a3477397d100a02eb618371697c7f3ee4e143e2c')
@@ -43,11 +44,15 @@ class Ucx(AutotoolsPackage):
             description='Enable assertions')
     variant('parameter_checking', default=False,
             description='Enable paramter checking')
-    variant('pic', default=False,
+    variant('pic', default=True,
             description='Builds with PIC support')
+    variant('java', default=False,
+            description='Builds with Java bindings')
 
     depends_on('numactl')
     depends_on('rdma-core')
+    depends_on('java@8', when='+java')
+    depends_on('maven', when='+java')
 
     def configure_args(self):
         spec = self.spec
@@ -81,5 +86,11 @@ class Ucx(AutotoolsPackage):
             config_args.append('--with-pic')
         else:
             config_args.append('--without-pic')
+
+        if '+java' in spec:
+            config_args.append('--with-java=%s' % spec['java'].prefix)
+        else:
+            config_args.append('--without-java')
+
 
         return config_args

--- a/var/spack/repos/builtin/packages/ucx/package.py
+++ b/var/spack/repos/builtin/packages/ucx/package.py
@@ -92,5 +92,4 @@ class Ucx(AutotoolsPackage):
         else:
             config_args.append('--without-java')
 
-
         return config_args

